javascript/src/musica_addon.cpp

```diff
@@ -0,0 +1,57 @@
+#include <napi.h>
+#include "musica_wrapper.h"
+#include <memory>
+#include <musica/version.hpp>
+
+
+using namespace musica_addon;
+
+// Global wrapper instance for static functions
+static std::unique_ptr<MusicaWrapper> g_wrapper;
+
+// Simple functions that match the stub interface
+Napi::String GetVersion(const Napi::CallbackInfo& info) {
+    Napi::Env env = info.Env();
+    try {
+        if (!g_wrapper) {
+            g_wrapper = std::make_unique<MusicaWrapper>();
+        }
+        return Napi::String::New(env, g_wrapper->GetVersion());
+    } catch (const std::exception& e) {
+        return Napi::String::New(env, "Error: " + std::string(e.what()));
+    }
+}
+
+Napi::Object GetSystemInfo(const Napi::CallbackInfo& info) {
+    Napi::Env env = info.Env();
+    Napi::Object obj = Napi::Object::New(env);
+
+    obj.Set("platform", Napi::String::New(env, "darwin"));
+    obj.Set("arch", Napi::String::New(env, "arm64"));
+    obj.Set("nodeVersion", Napi::String::New(env, "20.17.0"));
+
+    try {
+        if (!g_wrapper) {
+            g_wrapper = std::make_unique<MusicaWrapper>();
+        }
+        obj.Set("musicaVersion", Napi::String::New(env, g_wrapper->GetVersion()));
+    } catch (const std::exception& e) {
+        obj.Set("musicaVersion", Napi::String::New(env, "Error"));
+    }
```

Comment:
Could you include the error message?
```suggestion
    } catch (const std::exception& e) {
        obj.Set("musicaVersion", Napi::String::New(env, std::string("Error: ") + e.what()));
    }
```

Maybe it would make more sense to just throw an error?
```C++
throw Napi::Error::New(env, e.what());
```